src/ipc/handler.c

```diff
@@ -747,6 +747,47 @@ static int ipc_dma_trace_config(uint32_t header)
 	return err;
 }
 
+static int ipc_dma_trace_filter_update(uint32_t header)
+{
+	struct sof_ipc_trace_filter_elem *next_elem;
+	struct sof_ipc_trace_filter_elem *elem;
+	struct sof_ipc_trace_filter_elem *end;
+	struct sof_ipc_trace_filter *packet;
+	struct trace_filter filter;
+	struct ipc *ipc = ipc_get();
+	int ret = 0;
+	int cnt;
+
+	packet = ipc->comp_data;
+	elem = packet->elems;
+	end = &packet->elems[packet->elem_cnt];
```

Comment:
Can we validate end and elem_cnt is not overflowing msg size and mailbox.

Reply:
added